l1-contracts/contracts/state-transition/TestnetVerifier.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {IVerifier} from "./chain-interfaces/IVerifier.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @notice Modified version of the main verifier contract for the testnet environment
+/// @dev This contract is used to skip the zkp verification for the testnet environment.
+/// If the proof is not empty, it will verify it using the main verifier contract,
+/// otherwise, it will skip the verification.
+contract TestnetVerifier is IVerifier {
```

Comment:
Do you think it would be easier to make it inherit from the Verifier and call `super.verify` in case the proof length is non zero?

With the current design, we'll always have to deploy the "correct" verifier on L1 for testnet also, which is an unneeded additional step.

Reply:
That would involve changing the `verify` function visibility from `external` to `public` and adding the `virtual` keyword in order to override the method.

I think this proposed solution is more elegant than the current one, although I'm not sure if the function signature changes will affect anything else.

Reply:
Function signature won't change (signature is formula is hash of method name and it's parameters, but not other keywords (public/external/override and etc) and I like Stas's solution.

Reply:
Perfect, implemented it on 9367168c3665ce2bcf703bd1f8c65e905274af81